daml-lf/engine/src/main/scala/com/digitalasset/daml/lf/engine/Engine.scala

```diff
@@ -428,31 +435,17 @@ class Engine(val config: EngineConfig = new EngineConfig(LanguageVersion.StableV
       _ <- pkgs
         .collectFirst {
           case (pkgId, pkg) if !config.allowedLanguageVersions.contains(pkg.languageVersion) =>
-            Error.Package.Generic(
-              s"Disallowed language version in package $pkgId: " +
-                s"Expected version between ${config.allowedLanguageVersions.min.pretty} and ${config.allowedLanguageVersions.max.pretty} but got ${pkg.languageVersion.pretty}"
+            Error.Package.AllowedLanguageVersion(
+              pkgId,
+              pkg.languageVersion,
+              config.allowedLanguageVersions,
             )
         }
         .toLeft(())
-      _ <- {
-        val unknownPackages = pkgIds.filterNot(pkgs.isDefinedAt)
-        Either.cond(
-          unknownPackages.isEmpty,
-          (),
-          Error.Package.Generic(s"Unknown packages ${unknownPackages.mkString(", ")}"),
-        )
-      }
-      _ <- {
-        val missingDeps = pkgs.valuesIterator.flatMap(_.directDeps).toSet.filterNot(pkgIds)
-        Either.cond(
-          missingDeps.isEmpty,
-          (),
-          Error.Package.Generic(
-            s"The set of packages ${pkgIds.mkString("{'", "', '", "'}")} is not self consistent, the missing dependencies are ${missingDeps
-              .mkString("{'", "', '", "'}")}."
-          ),
-        )
-      }
+      unknownPackages = pkgIds.filterNot(pkgs.isDefinedAt)
```

Comment:
Looking at the usages of this function I’m not entirely sure the API here makes sense. We pass in a `Set[PackageId]` and a `Map[PackageId, Package]` and then assert that the set contains the key set of the map. We should be able to just accept the map only, eliminate this error and afaict all callsites would work well with that. Probably something for a separate PR though.

Reply:
https://github.com/digital-asset/daml/pull/9986